File: leetcode-104_maximum-depth-of-binary-tree.go
/**
 * 给定一个二叉树, 求二叉树的最大高度
 */
func maxDepth(root *TreeNode) int {
    if root == nil {
        return 0
    }
    leftMaxDepth := maxDepth(root.Left) + 1
    rightMaxDepth := maxDepth(root.Right) + 1
    return int(math.Max(float64(leftMaxDepth), float64(rightMaxDepth)))
}

File: leetcode-104_maximum-depth-of-binary-tree.java
/**
 * 给定一个二叉树, 求二叉树的最大高度
 */
public int maxDepth(TreeNode root) {
    if (root == null) {
        return 0;
    }

    int leftDepth = maxDepth(root.left) + 1;
    int rightDepth = maxDepth(root.right) + 1;

    return Math.max(leftDepth, rightDepth);
}
